### PR TITLE
refactor: remove hardcoded things [AR-1436][AR-1343][AR-1453][AR-1410]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1436" title="AR-1436" target="_blank">TaskAR-1436</a>  Remove hardcoded statuses and indicators
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

There were some UI elements hardcoded in the app that needed to be hidden or refactored to use the real data instead of mock:
- Legal hold indicator
- "real conversations"
- Unread badges
- Status indicators (available, busy)
- Second Team in profile
- Team ID ↔︎ Team name
- Profile Image in conversation screen (it’s always green)
- Profile image when conversation is long pressed

Also small things from design review:
- Register flows: “Accept Terms” dialog: The title still reads “Terms & Conditions” (should be called “Terms of Use” instead)
- Links to the Terms of Use and Pricing lead to a dead page (404)

### Solutions

- Legal hold indicator - now returned from kalium even for group conversations, fetching this data is not yet implemented in kalium
- "real conversations" - header changed to the proper one according to designs
- unread badges - now showing the real values (for mentions and calls it's still mock data but correctly calculated)
- status indicators - now taken from ViewModel, fetching user statuses is not yet implemented in kalium, but from UI perspective it's ready
- Second Team in profile - removed hardcoded data, fetching this data is not yet implemented in kalium
- Team ID ↔︎ Team name - now fetched using proper use case and shown as it should be
- Profile Image in conversation screen - now fetching is properly implemented
- Profile image when conversation is long pressed - now fetching is properly implemented
- Register flows - "Terms of Use" changed
- Links to the Terms of Use and Pricing lead to a dead page - fixed by removing duplicated "https://"

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Just open the app and look at all the listed places if elements are as they should be.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-544/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-544/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3211935)(3211935)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3211935)(3211935)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-544/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3219692)(3219692)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3219692)(3219692)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-544/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3225408)(3225408)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3225408)(3225408)-dev-release.apk)

**Review by MohamadJaara — APPROVED** [review]
lgtm ✅

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt [hunk 1]
```diff
@@ -161,11 +170,11 @@ class ConversationListViewModel @Inject constructor(
...
                         conversationInfo = ConversationInfo(
                             name = otherUser.name.orEmpty(),
-                            membership = mapUserType(details.userType),
-                            isLegalHold = false
+                            membership = mapUserType(details.userType)
                         ),
                         conversationId = conversation.id,
-                        mutedStatus = conversation.mutedStatus
+                        mutedStatus = conversation.mutedStatus,
+                        isLegalHold = details.legalHoldStatus == LegalHoldStatus.ENABLED
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt:177]: Can you extract details.legalHoldStatus == LegalHoldStatus.ENABLED to a separate function nstead of checking it everywhere ?

**Review by ohassine — APPROVED** [review]
Nice to see all of these changes 🚀 Good job ✌️, I just left two small comments

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt [hunk 2]
```diff
@@ -101,7 +103,17 @@ class ConversationViewModel @Inject constructor(
                     is ConversationDetails.OneOne -> conversationDetails.otherUser.name.orEmpty()
                     else -> conversationDetails.conversation.name.orEmpty()
                 }
-                conversationViewState = conversationViewState.copy(conversationName = conversationName)
+                val conversationAvatar = when (conversationDetails) {
+                    is ConversationDetails.OneOne ->
+                        ConversationAvatar.OneOne(conversationDetails.otherUser.previewPicture?.let { UserAvatarAsset(it) })
+                    is ConversationDetails.Group ->
+                        ConversationAvatar.Group(getConversationColor(conversationDetails.conversation.id))
+                    else -> ConversationAvatar.None
```
[the remark this answers is not in the document]
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:111]: Ah my bad, it can be SELF...Ignore my comment 🙂

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-544/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3226084)(3226084)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3226084)(3226084)-dev-release.apk)

**Pull request merged**